gpytorch/likelihoods/gaussian_likelihood.py

```diff
@@ -1,42 +1,45 @@
 from __future__ import absolute_import, division, print_function, unicode_literals
 
 import math
-import torch
-from ..distributions import MultivariateNormal
-from ..functions import add_diag
-from ..likelihoods import Likelihood
-from .. import settings
+import warnings
+import logging
 
+from .. import settings
+from ..distributions import MultivariateNormal
+from ..lazy import AddedDiagLazyTensor, DiagLazyTensor
+from .likelihood import Likelihood
+from .noise_models import HomoskedasticNoise
 
-class GaussianLikelihood(Likelihood):
-    r"""
-    """
-
-    def __init__(self, log_noise_prior=None, batch_size=1):
-        super(GaussianLikelihood, self).__init__()
-        self.register_parameter(
-            name="log_noise", parameter=torch.nn.Parameter(torch.zeros(batch_size, 1)), prior=log_noise_prior
-        )
 
-    @property
-    def noise(self):
-        return self.log_noise.exp()
+class _GaussianLikelihoodBase(Likelihood):
+    def __init__(self, log_noise_covar):
+        super(_GaussianLikelihoodBase, self).__init__()
+        self.log_noise_covar = log_noise_covar
 
-    def forward(self, input):
+    def forward(self, input, *params):
         if not isinstance(input, MultivariateNormal):
-            raise ValueError("GaussianLikelihood requires a MultivariateNormal input")
+            raise ValueError("Gaussian Likelihoods require a MultivariateNormal input")
         mean, covar = input.mean, input.lazy_covariance_matrix
-        noise = self.noise
-        if covar.ndimension() == 2:
-            if settings.debug.on() and noise.size(0) > 1:
-                raise RuntimeError("With batch_size > 1, expected a batched MultivariateNormal distribution.")
-            noise = noise.squeeze(0)
+        log_noise_covar = self.log_noise_covar(*params)
```

Comment:
I would imagine we may want `log_noise_covar` modules to actually be just `noise_covar` modules, because most forms other than diagonal will probably have some other strategy to ensure positive definiteness (e.g., maybe it returns a `CholLazyTensor`). The standard one can just exponentiate by itself.